porch/pkg/engine/clone.go

```diff
@@ -89,14 +89,19 @@ func (m *clonePackageMutation) cloneFromRegisteredRepository(ctx context.Context
 		return repository.PackageResources{}, fmt.Errorf("cannot read contents of package %q: %w", ref.Name, err)
 	}
 
+	// If the upstream we cloned from has its own upstream information, we need to clear and replace it
+	if err := kpt.UpdateKptfileUpstream(m.name, resources.Spec.Resources, v1.Upstream{}, v1.UpstreamLock{}); err != nil {
+		return repository.PackageResources{}, fmt.Errorf("failed to clear upstream lock to package %q: %w", ref.Name, err)
+	}
+
 	upstream, lock, err := revision.GetUpstreamLock()
 	if err != nil {
 		return repository.PackageResources{}, fmt.Errorf("cannot determine upstream lock for package %q: %w", ref.Name, err)
 	}
 
 	// Update Kptfile
 	if err := kpt.UpdateKptfileUpstream(m.name, resources.Spec.Resources, upstream, lock); err != nil {
-		return repository.PackageResources{}, fmt.Errorf("failed to apply upstream lock to pakcage %q: %w", ref.Name, err)
+		return repository.PackageResources{}, fmt.Errorf("failed to apply upstream lock to package %q: %w", ref.Name, err)
```

Comment:
Was trying to trace this logic for three level deep inheritance and wasn't sure about this logic. Can you pl. run through the following scenario. Happy to discuss offline if needed.

```
P1 (originated in porch, so no upstream)

P1' (cloned from P1) ==> upstream(P1') should point to the git commit corresponding to the publishing of P1. 

P1'' (cloned from P1') ==> upstream(P1'') should point to the git commit corresponding to the publishing of P1'.
```

Reply:
> Lines 92-95 of this file should clear them. Then, porch makes a call to GetUpstreamLock() which should fail to find the information in the Kptfile because we cleared it, and should look at the parent pointer (pointing back to P1', where we cloned from).

Aah, I see it now. I was confused about why we are resetting the upstream info to empty (line 93) if we are going to update the upstream anyways at line 103 anyways. So, we are resetting the upstream here so that the GetUpstreamLock() can return the commit of the parent. Reading your comment again, I realize that we always want the parent commit, then why not just get the parent info and use that in the Kptfile ?

Reply:
Parent does not always point to the upstream (I give an example in my PR description above with the edit/copy task)

Since the correct upstream information is _always_ available in the Kptfile with the only exception being a fresh clone, I am treating the Kptfile as the source of truth with the only exception being the clone task, which will get the upstream information from the parent. 

If we only look at the parent in GetUpstreamLock (which was the previous implementation), we’ll need a different method to look at the Kptfile when we do an update task because parent is not reliable. I thought about doing that, but it seemed weird to me to have a method called GetUpstreamLock that wasn’t actually getting the upstream in all cases, which is why I changed it. 

I’m happy to restructure the code somehow but looking at just the parent doesn’t seem to be the right thing to do.

Reply:
As discussed offline, let's capture the difference between `editing a package` and `cloning package` as a comment and move on till we have a better idea to resolve it.

Reply:
Thanks! I added the comment. Will merge once the tests pass.